v8go.cc

```diff
@@ -1168,14 +1168,24 @@ static void buildCallArguments(Isolate* iso, Local<Value> *argv, int argc, Value
   }
 }
 
-RtnValue FunctionCall(ValuePtr ptr, int argc, ValuePtr args[]) {
+RtnValue FunctionCall(ValuePtr ptr, ValuePtr recv, int argc, ValuePtr args[]) {
   LOCAL_VALUE(ptr)
+
   RtnValue rtn = {nullptr, nullptr};
   Local<Function> fn = Local<Function>::Cast(value);
   Local<Value> argv[argc];
   buildCallArguments(iso, argv, argc, args);
-  Local<Value> recv = Undefined(iso);
-  MaybeLocal<Value> result = fn->Call(local_ctx, recv, argc, argv);
+
+  MaybeLocal<Value> result;
+
+  if (recv == NULL) {
+    result = fn->Call(local_ctx, Undefined(iso), argc, argv);
```

Comment:
To maintain backwards compatibility for the API but actually correctly implement the `FunctionCall` for the v8 api, I was thinking of supporting either a nul or obj receiver and handling either case internally so the v8go function method does not need to change as part of this PR.